packages/eslint-plugin/src/rules/use-unknown-in-catch-callback-variable.ts

```diff
@@ -288,68 +237,75 @@ export default createRule<[], MessageIds>({
     }
 
     return {
-      CallExpression(node): void {
-        if (node.arguments.length === 0 || !isPromiseCatchAccess(node.callee)) {
+      CallExpression({ arguments: args, callee }): void {
+        if (callee.type !== AST_NODE_TYPES.MemberExpression) {
           return;
         }
 
-        const firstArgument = node.arguments[0];
+        const staticMemberAccessKey = getStaticMemberAccessKey(
+          callee,
+          context.sourceCode.getScope(callee),
```

Comment:
`scope` was always missing, meaning that `const mk = 'catch'; promise[mk]` was not being detected properly ([playground](https://typescript-eslint.io/play/#ts=5.5.2&fileType=.tsx&code=MYewdgzgLgBAHjAvDA5MAhlYALFBuAKAAUAnEAWwEsIBTAOhJohABsA3GgCgEoBtOALqdONEiQBcMdGACe3JAD4YAbxgBfboVIVq9Rs3Zc%2BaTDhRCRYydLmKV6zUA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYkACAXAngA4oDG0AlobgLQrzkB2uA9LMotbAwNYMD2AdwbVG1EgENcJABZjx8eACNxJbtQBu4iuMVJ0URNGh9okAL4gzQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false))